lib/spack/spack/cmd/develop.py

```diff
@@ -116,9 +116,15 @@ def develop(parser, args):
         raise SpackError("spack develop requires at most one named spec")
 
     spec = specs[0]
+
     version = spec.versions.concrete_range_as_version
     if not version:
-        raise SpackError("Packages to develop must have a concrete version")
+        # look up the maximum version so infintiy versions are preferred for develop
+        version = max(spec.package_class.versions.keys())
+        tty.warn(
+            f"{spec.name} was not given a version to develop with. "
+            f"Defaulting to the maximal version: {spec.name}@{version}"
+        )
```

Comment:
```suggestion
        tty.msg(
            f"Defaulting to highest version: {spec.name}@{version}"
        )
```
I don't think this should be so loud as a warning. It's in line with user expectations.